File: leetcode/findSubsequences.go
package leetcode

import "fmt"

func findSubsequences(nums []int) [][]int {

	var subs [][]int

	for _, num := range nums {
		var tmp [][]int
		tmp = append(tmp, []int{num})

		for i := 0; i < len(subs); i++ {
			if subs[i][len(subs[i])-1] <= num {
				tmp = append(tmp, subs[i])
				subs[i] = append(subs[i], num)
			}
		}
		subs = append(subs, tmp...)
	}
	fmt.Println(subs)

	var result [][]int

	for i := 0; i < len(subs); i++ {
		if len(subs[i]) >= 2 {
			result = append(result, subs[i])
		}
	}

	return result

}

File: leetcode/findSubsequences.py

class Solution(object):

    def findSubsequences(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        subs = {()}
        for num in nums:
            subs |= {sub + (num,)
                    for sub in subs
                    if not sub or sub[-1] <= num}
            print(subs)
        return [sub for sub in subs if len(sub) >= 2]
        
if __name__ == '__main__':
    print(Solution().findSubsequences([4, 6, 7, 7]))
